db_stress_tool/expected_value.h

```diff
@@ -160,7 +163,51 @@ class PendingExpectedValue {
         orig_value_(orig_value),
         final_value_(final_value) {}
 
+  PendingExpectedValue(const PendingExpectedValue& other)
+      : value_ptr_(other.value_ptr_),
+        orig_value_(other.orig_value_),
+        final_value_(other.final_value_) {
+    other.ClosePendingState();
+  }
+
+  PendingExpectedValue(PendingExpectedValue&& other)
+      : value_ptr_(std::move(other.value_ptr_)),
+        orig_value_(std::move(other.orig_value_)),
+        final_value_(std::move(other.final_value_)) {
+    other.ClosePendingState();
+  }
+
+  PendingExpectedValue& operator=(const PendingExpectedValue& other) {
+    if (this != &other) {
+      other.ClosePendingState();
+      value_ptr_ = other.value_ptr_;
+      orig_value_ = other.orig_value_;
+      final_value_ = other.final_value_;
+    }
+    return *this;
+  }
+
+  PendingExpectedValue& operator=(PendingExpectedValue&& other) {
+    if (this != &other) {
+      other.ClosePendingState();
+      value_ptr_ = std::move(other.value_ptr_);
+      orig_value_ = std::move(other.orig_value_);
+      final_value_ = std::move(other.final_value_);
+    }
+    return *this;
+  }
+
+  ~PendingExpectedValue() {
+    if (!pending_state_closed_) {
+      fprintf(stderr,
+              "A PendingExpectedValue is still pending at destruction %p\n",
+              this);
+    }
```

Comment:
Since it's part of our testing infra, we can assert instead

Reply:
Good point! Updated.